api/src/main/java/org/openmrs/module/ugandaemrreports/reports/SetupDueForViralLoadReport.java

```diff
@@ -0,0 +1,148 @@
+        return "Due For Viral Load";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Due For Viral Load";
+    }
+
+    @Override
+    public List<Parameter> getParameters() {
+        List<Parameter> l = new ArrayList<Parameter>();
+        l.add(df.getStartDateParameter());
+        l.add(df.getEndDateParameter());
+        return l;
+    }
+
+    @Override
+    public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+        List<ReportDesign> l = new ArrayList<ReportDesign>();
+        l.add(buildReportDesign(reportDefinition));
+        return l;
+    }
+
+    /**
+     * Build the report design for the specified report, this allows a user to override the report design by adding
+     * properties and other metadata to the report design
+     *
+     * @param reportDefinition
+     * @return The report design
+     */
+    @Override
+
+    public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+        ReportDesign rd = createExcelTemplateDesign(getExcelDesignUuid(), reportDefinition, "DueForViralLoad.xls");
+        Properties props = new Properties();
+        props.put("repeatingSections", "sheet:1,row:8,dataset:DUE_FOR_VIRAL_LOAD");
+        props.put("sortWeight", "5000");
+        rd.setProperties(props);
+        return rd;
+    }
+
+    @Override
+    public ReportDefinition constructReportDefinition() {
+
+        ReportDefinition rd = new ReportDefinition();
+
+        rd.setUuid(getUuid());
+        rd.setName(getName());
+        rd.setDescription(getDescription());
+        rd.setParameters(getParameters());
+
+        PatientDataSetDefinition dsd = new PatientDataSetDefinition();
+
+        CohortDefinition patientsDueForViralLoad = df.getPatientsDueForViralLoad(hivMetadata.getReturnVisitDate(), Arrays.asList(hivMetadata.getARTEncounterEncounterType()), BaseObsCohortDefinition.TimeModifier.ANY);
+
+        dsd.setName(getName());
+        dsd.setParameters(getParameters());
+        dsd.addRowFilter(Mapped.mapStraightThrough(patientsDueForViralLoad));
+        addColumn(dsd, "Clinic No", hivPatientData.getClinicNumber());
+        dsd.addColumn( "Patient Name",  new PreferredNameDataDefinition(), (String) null);
+        dsd.addColumn( "Sex", new GenderDataDefinition(), (String) null);
+        dsd.addColumn("Birth Date", builtInPatientData.getBirthdate(), "", new BirthDateConverter());
+        addColumn(dsd, "Age", builtInPatientData.getAgeAtStart());
+        addColumn(dsd, "HIV Enrolled Date", hivPatientData.getEnrollmentDate());
+        addColumn(dsd, "ART Start Date", hivPatientData.getArtStartDate());
+        addColumn(dsd, "Viral Load Date", hivPatientData.getViralLoadDate());
+        addColumn(dsd, "Viral Load Qualitative", hivPatientData.getViralLoadQualitative());
+        addColumn(dsd, "Viral Load", hivPatientData.getCurrentViralLoad());
+        addColumn(dsd, "Last Visit Date", hivPatientData.getLastVisitDate());
+
```

Comment:
@solemabrothers no need for the blank lines

Reply:
@ssmusoke this has also been fixed